homeassistant/core.py

```diff
@@ -637,30 +671,27 @@ async def async_block_till_done(self) -> None:
         """Block until all pending work is done."""
         # To flush out any call_soon_threadsafe
         await asyncio.sleep(0)
+        await asyncio.sleep(0)
         start_time: float | None = None
-
-        while self._pending_tasks:
-            pending = [task for task in self._pending_tasks if not task.done()]
-            self._pending_tasks.clear()
```

Comment:
The old approach would clear all the tasks so it would only track _new_ tasks. The new approach is to just track all tasks. Old + new. It doesn't matter because if the old one blocks, it's also blocking.